test/shared/Pair.js

```diff
@@ -0,0 +1,45 @@
+var R = require('../../source');
+
+
+function Pair(left, right) {
+  return {
+    '@@type': 'ramda/Pair',
+    'fantasy-land/equals': function equals(pair) {
+      return pair != null
+        && pair['@@type'] === this['@@type']
+        && R.equals(left, pair.left)
+        && R.equals(right, pair.right);
+    },
+    'fantasy-land/concat': function concat(pair) {
+      return Pair(R.concat(left, pair.left), R.concat(right, pair.right));
+    },
+    'fantasy-land/map': function map(f) { return Pair(left, f(right)); },
+    'fantasy-land/ap': function ap(pair) {
+      return Pair(R.concat(left, pair.left), right(pair.right));
+    },
+    'fantasy-land/reduce': function reduce(f, x) { return f(x, right); },
+    'fantasy-land/traverse': function traverse(of, f) {
+      return R.map(
+        function(x) { return Pair(left, x); },
+        of(f(right))
+      );
+    },
+    'fantasy-land/chain': function chain(f) {
+      var pair = f(right);
+      return Pair(R.concat(left, pair.left), pair.right);
+    },
+    'fantasy-land/bimap': function bimap(f, g) { return Pair(f(left), g(right)); },
+    'fantasy-land/promap': function promap(f, g) {
+      return Pair(R.promap(f, g, left), R.promap(f, g, right));
+    },
+    merge: function merge(f) { return f(left, right); },
+    toString: function toString() {
+      return `Pair(${R.toString(left)}, ${R.toString(right)})`;
+    },
+    toArray: function toArray() { return [left, right]; },
+    left: left,
+    right: right,
+  };
+}
+
+module.exports = Pair;
```

Comment:
Another option is to depend on [sanctuary-pair][1].


[1]: https://github.com/sanctuary-js/sanctuary-pair

Reply:
It's hard to find a point with small changes where it's worth introducing a first dependency into a dependency-free library.  With a large change (say sanctuary-def) it would be a very different matter.

Reply:
For what it's worth (probably not much ;) I do appreciate libraries that make a conscious effort to keep their dependency tree to a minimum. Ramda has none and it's truly _amazing_. I'm sure that must be a contributing factor when it comes to adopting a library. (Admittedly not the most important one nor the only one neither.)

Reply:
It would only be a development dependency, of which Ramda already has many.

Reply:
Ah sorry @davidchambers I didn't realise this was in a test file!

Reply:
I wasn't noticing that either.  I don't object to a dependency here.  Nor do I feel any great need.

Reply:
I'm not sure that I can use santuary-pair in this case since I wanted to provide the Profunctor instance of Pair for the `fantasy-land/promap` dispatch test case, which it doesn't look like I'd get from sanctuary-pair (unless there's a standard mechanism for providing such implementations to Sanctuary offerings). I chose Pair in this case because it's the simplest non-Function Profunctor instance I could think of 😅